pkg/services/ngalert/api/api_alertmanager.go

```diff
@@ -0,0 +1,362 @@
+				InhibitedBy: []string{"inhibitedBy 1"},
+				SilencedBy:  []string{"silencedBy 1"},
+				State:       stringPtr(amv2.AlertStatusStateActive),
+			},
+			UpdatedAt: timePtr(strfmt.DateTime(now.Add(-time.Hour))),
+			Alert: amv2.Alert{
+				GeneratorURL: strfmt.URI("a URL"),
+				Labels: amv2.LabelSet{
+					"label1-1": "value1",
+					"label1-2": "value2",
+				},
+			},
+		},
+		&amv2.GettableAlert{
+			Annotations: amv2.LabelSet{
+				"annotation2-1": "value1",
+				"annotation2-2": "value2",
+			},
+			EndsAt:      timePtr(strfmt.DateTime(now.Add(time.Hour))),
+			Fingerprint: stringPtr("fingerprint 2"),
+			Receivers: []*amv2.Receiver{
+				{
+					Name: stringPtr("receiver identifier 2-1"),
+				},
+				{
+					Name: stringPtr("receiver identifier 2-2"),
+				},
+			},
+			StartsAt: timePtr(strfmt.DateTime(now)),
+			Status: &amv2.AlertStatus{
+				InhibitedBy: []string{"inhibitedBy 2"},
+				SilencedBy:  []string{"silencedBy 2"},
+				State:       stringPtr(amv2.AlertStatusStateActive),
+			},
+			UpdatedAt: timePtr(strfmt.DateTime(now.Add(-time.Hour))),
+			Alert: amv2.Alert{
+				GeneratorURL: strfmt.URI("a URL"),
+				Labels: amv2.LabelSet{
+					"label2-1": "value1",
+					"label2-2": "value2",
+				},
+			},
+		},
+	}
+	return response.JSON(http.StatusOK, result)
+}
+
+func (srv AlertmanagerSrv) RouteGetSilence(c *models.ReqContext) response.Response {
+	silenceID := c.Params(":SilenceId")
+	srv.log.Info("RouteGetSilence: ", "SilenceId", silenceID)
+	q := ngmodels.GetSilenceByUIDQuery{OrgID: c.SignedInUser.OrgId, UID: silenceID}
+	if err := srv.store.GetSilenceByUID(&q); err != nil {
+		if errors.Is(err, ngmodels.ErrSilenceNotFound) {
+			return response.Error(http.StatusNotFound, "silence not found", err)
+		}
+		return response.Error(http.StatusInternalServerError, "failed to get silence", err)
+	}
+
+	return response.JSON(http.StatusOK, q.Result.ToGettableSilence())
+}
+
+func (srv AlertmanagerSrv) RouteGetSilences(c *models.ReqContext) response.Response {
+	q := ngmodels.GetSilencesQuery{OrgID: c.SignedInUser.OrgId}
+	if err := srv.store.GetOrgSilences(&q); err != nil {
+		return response.Error(http.StatusInternalServerError, "failed to get silences", err)
+	}
+	result := apimodels.GettableSilences{}
+	for _, s := range q.Result {
+		gettableSilence := s.ToGettableSilence()
+		result = append(result, &gettableSilence)
+	}
+	return response.JSON(http.StatusOK, result)
+}
+
+func (srv AlertmanagerSrv) RoutePostAlertingConfig(c *models.ReqContext, body apimodels.PostableUserConfig) response.Response {
+	config, err := yaml.Marshal(&body)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "failed to serialize to the Alertmanager configuration", err)
+	}
+
+	err = srv.store.SaveAlertmanagerConfiguration(&ngmodels.SaveAlertmanagerConfigurationCmd{
+		AlertmanagerConfiguration: string(config),
+		ConfigurationVersion:      fmt.Sprintf("v%d", ngmodels.AlertConfigurationVersion),
+	})
+
+	if err != nil {
```

Comment:
Nit, but makes for better reading if error checks are co-located with the assignment.
```suggestion
	if err != nil {
```